### PR TITLE
Increase Beacon client code coverage

### PR DESCRIPTION
**Pull request by claravanstaden**
Adds tests for cases described in Linear issue.

Resolves: [SNO-562](https://linear.app/snowfork/issue/SNO-562)

### CODE REVIEW

File: parachain/pallets/ethereum-beacon-client/src/tests.rs [hunk 1]
```diff
@@ -36,23 +42,6 @@ pub fn compute_domain() {
...
-		);
-
-		assert_ok!(&domain);
-		assert_eq!(
-			domain.unwrap(),
-			hex!("07000000e7acb21061790987fa1c1e745cccfb358370b33e8af2b2c18938e6c2").into()
-		);
-	});
-}
-
```
Comment (claravanstaden) [inline parachain/pallets/ethereum-beacon-client/src/tests.rs:55]: Kiln specific data tests aren't necessary anymore.

**Comment by claravanstaden** [comment]
Covers all the tests discussed in [SNO-562](https://linear.app/snowfork/issue/SNO-562), except for `impls.rs` (will be added in a new PR). Coverage increased from `85.98%` to `94.41%`: https://app.codecov.io/gh/Snowfork/snowbridge/blob/clara%2Fsno-562/parachain%2Fpallets%2Fethereum-beacon-client%2Fsrc%2Flib.rs

**Review by alistair-singh — APPROVED** [review]

**Review by yrong — APPROVED** [review]
Awesome!

**Review by vgeddes — APPROVED** [review]
+1

**Pull request merged**